src/NATS.Client.Core/Internal/SslStreamConnection.cs

```diff
@@ -170,6 +170,42 @@ private SslClientAuthenticationOptions SslClientAuthenticationOptions(NatsUri ur
             rcsCb = RcsCbCaCertChain;
         }
 
+#if NET8_0_OR_GREATER
+        X509ChainPolicy? policy = null;
+        SslStreamCertificateContext? streamCertificateContext = null;
+        if (_tlsCerts?.ClientCerts != null && _tlsCerts.CaCerts != null && _tlsCerts.ClientCerts.Count >= 1)
```

Comment:
Why check for `_tlsCerts.CaCerts != null` here?  Shouldn't CA Certs be handled separately, if they aren't set here then check the OS store.  Could still have a client cert with intermediates and a CA Cert in the OS store.

Reply:
good point. Also if CA is in the list of client certs, would we still need to provide CA separately (if not is OS store)?

Reply:
The CA list is for validating the Server's certs, so we want the distinction between Client Certs and CA Certs and should not cross them.  The Client and Server certs could technically be signed by different trust roots.

Reply:
then don't we need clientCAs + serverCAs?

Reply:
I think there is no such thing as a Client CA really...  just a Client Cert + optional Client intermediaries

Reply:
Its a good callout that the root CA may not be the same for client chain and server chain. @caleblloyd / @mtmk is it a fair assumption that 
1. all behavior around server certificate chain validation will occur in `RcsCbCaCertChain`?
2. Server will always have knowledge of the client chain's root CA?

Reply:
My understanding is "yes" to both of those assumptions

Reply:
OK, got it! So client bundle would include it's root CA for the client. So `SslStreamCertificateContext` doesn't need to know about the `caCert` in our settings at all since our `caCert` is for the server chain validation. and yes server will have to trust at least one of the client's CAs, typically root.